i18n(pt-BR): Updated `guides/typescript.mdx` (#3953)
withastro/docs: src/content/docs/pt-br/guides/typescript.mdx

}
```

> ## Extendendo `window` e `globalThis`

Reviewer: Aqui deveria ser "estender" para se adequar a forma normativa.
```suggestion
## Estendendo `window` e `globalThis`
```

Você pode querer adicionar uma propriedade para o objeto global. Você pode fazer isso adicionando uma declaração global ao seu arquivo `env.d.ts`:

```ts title="env.d.ts"
declare global {
	var minhaString: string;
	function minhafuncao(): boolean;
}

export {};
```

Isso vai fornecer tipos para `globalThis.minhaString` e `globalThis.minhaFuncao`, assim como para `window.minhaString` e `window.minhaFuncao`.

Note que `window` só está disponível no código no lado do cliente. O `globalThis` está disponível no lado do servidor e no lado do cliente, mas seu valor no lado do servidor não é compartilhado com o cliente.

> Se você só quer definir o tipo de uma propriedade no objeti `window`, forneça uma interface `Window` ao invés disso:

Reviewer: ```suggestion
Se você só quer definir o tipo de uma propriedade no objeto `window`, forneça uma interface `Window` ao invés disso:
```

<h2>{saudacao}, {nome}!</h2>
```

### Padrões de tipo de prop comuns

- Se seu componente nao recebe props ou conteúdo em slot, você pode usar `type Props = Record<string, never>`.

> - Se seu componente precisa ser passado filhos para seu slot padrão, você pode impor isso usando `type Props = { children: any; };`.

Reviewer: ```suggestion
- Se seu componente precisa passar filhos para seu slot padrão, você pode impor isso usando `type Props = { children: any; };`.
```

Author: Aqui eu fiquei confuso por que a frase está se referindo ao componente obrigar que passem filhos a ele, e não que meu componente passe filhos em algum lugar

Reviewer: Certo, eu tinha inferido sem olhar o original essa parte, talvez assim faça mais sentido?
````suggestion
- Se seu componente deve passar filhos para seu slot padrão, você pode impor isso usando `type Props = { children: any; };`.
```

Author: Não sei ainda parece estranho dizer que o componente passa filhos. Eu acho que é no sentido do componente receber filhos no slot padrão

Reviewer: Ah sim, consigo ver como alguém interpretaria assim, mudando o verbo acredito que a gente resolve essa confusão de vez.
```suggestion
- Se seu componente deve receber filhos em seu slot padrão, você pode impor isso usando `type Props = { children: any; };`.
```

## Utilitários de Tipagem

<Since v="1.6.0" />

O Astro vem com alguns tipos utilitários integrados para padrões de tipo de props comuns. Eles estão disponíveis dentro do entrypoint `astro/types`.

### Atributos HTML integrados

> O Astro fornece os tipos `HTMLAttributes` para checar que sua marcação está usando atributos HTML válidos. Você pode usar esses tipos para ajudar construir as props de componentes.

Reviewer: ```suggestion
O Astro fornece os tipos `HTMLAttributes` para checar que sua marcação está usando atributos HTML válidos. Você pode usar esses tipos para ajudar a construir as props de componentes.
```

> Por exemplo, se você está construindo um componente `<Link>`, você pode fazer o seguinte para espelhar os atributos padrão para tags `<a>` nos tipos da prop do seu componente.

Reviewer: ```suggestion
Por exemplo, se você está construindo um componente `<Link>`, você pode fazer o seguinte para espelhar os atributos padrão de tags `<a>` nos tipos da prop do seu componente.
```

```astro title="src/components/Link.astro" ins="HTMLAttributes" ins="HTMLAttributes<'a'>"
---
import { HTMLAttributes } from 'astro/types'
// use um `type`
type Props = HTMLAttributes<'a'>;

> // ou extenda com uma `interface`

Reviewer: ```suggestion
// ou estenda com uma `interface`
```

```
:::

> ### Tipo polimófico

Reviewer: Errinho de digitação
```suggestion
### Tipo polimórfico
```

<Since v="2.5.0" />

O Astro inclui um utilitário para tornar mais fácil construir componentes que podem renderizar como elementos HTML diferentes com segurança total de tipo. Isso é útil para componentes como `<Link>` que podem renderizar como `<a>` ou `<button>` dependendo das props passadas para ele.

> O exemplo abaixo implementa um compomente totalmente tipado e polimófico que pode renderizar como qualquer elemento HTML. O tipo [`HTMLTag`](#atributos-html-integrados) é usado para garantir que a prop `as` é um elemento HTML válido.

Reviewer: ```suggestion
O exemplo abaixo implementa um componente totalmente tipado e polimórfico que pode renderizar como qualquer elemento HTML. O tipo [`HTMLTag`](#atributos-html-integrados) é usado para garantir que a prop `as` é um elemento HTML válido.
```

```astro
---
import { HTMLTag, Polymorphic } from 'astro/types';

type Props<Tag extends HTMLTag> = Polymorphic<{ as: Tag }>;

const { as: Tag, ...props } = Astro.props;
---

<Tag {...props} />
```

### Inferir os tipos do `getStaticPaths()`

<Since v="2.1.0" />

> O Astro inclui utilitários para trabalhar com os tipos retornados pelas suas funções [`getStaticPaths()`](/pt-br/reference/api-reference/#getstaticpaths) para rotas dinâmicas.

Reviewer: ```suggestion
O Astro inclui utilitários para trabalhar com os tipos retornados pela sua função [`getStaticPaths()`](/pt-br/reference/api-reference/#getstaticpaths) para rotas dinâmicas.
```
